go-controller/pkg/testing/util.go

```diff
@@ -26,12 +26,36 @@ func GenerateNAD(networkName, name, namespace, topology, cidr, role string) *nad
 		cidr,
 		fmt.Sprintf("%s/%s", namespace, name),
 		role,
-	)
+	))
+}
+func GenerateNADWithType(networkName, name, namespace, topology, cidr, role, ty string) *nadapi.NetworkAttachmentDefinition {
```

Comment:
this function is not used?

Reply:
Yes, now that we know that the problem if with all the errors and not only when the type is unexpected we can remove this.